src/cleanup_queue.cc

```diff
@@ -8,27 +8,22 @@ namespace node {
 std::vector<CleanupQueue::CleanupHookCallback> CleanupQueue::GetOrdered()
     const {
   // Copy into a vector, since we can't sort an unordered_set in-place.
-  std::vector<CleanupHookCallback> callbacks(cleanup_hooks_.begin(),
-                                             cleanup_hooks_.end());
+  std::vector callbacks(cleanup_hooks_.begin(), cleanup_hooks_.end());
   // We can't erase the copied elements from `cleanup_hooks_` yet, because we
   // need to be able to check whether they were un-scheduled by another hook.
 
-  std::sort(callbacks.begin(),
-            callbacks.end(),
-            [](const CleanupHookCallback& a, const CleanupHookCallback& b) {
-              // Sort in descending order so that the most recently inserted
-              // callbacks are run first.
-              return a.insertion_order_counter_ > b.insertion_order_counter_;
-            });
+  // Sort in descending order so that the most recently inserted callbacks are
+  // run first.
+  std::ranges::sort(callbacks, std::greater());
 
   return callbacks;
 }
 
 void CleanupQueue::Drain() {
-  std::vector<CleanupHookCallback> callbacks = GetOrdered();
+  auto callbacks = GetOrdered();
```

Comment:
My impression on the https://github.com/nodejs/node/blob/main/doc/contributing/cpp-style-guide.md#using-auto is that only use `auto` in conditions like

1. the type has already been declared on the right hand side, e.g. `auto ptr = std::make_unique<MyType>()`,
2. the type is an iterator type,
3. the type is an RAII, e.g. `auto reset = OnScopeLeave(...)`.

Not a strong opinion but I think the rule is suggesting keeping the type explicit.